### ISSUE TITLE
Improve dangerouslySetInnerHTML

### ISSUE THREAD

**Issue by rebeccahum** [Automattic/VIP-Coding-Standards#267]
## What problem would the enhancement address for VIP?
Currently, the sniff for `dangerouslySetInnerHTML` flags any usage of it, even when it's being escaped.  

## Describe the solution you'd like
It would be nice to NOT flag for usage of `DOMPurify.sanitize()` in it since it is being escaped properly.

## What code should be reported as a violation?

```
dangerouslySetInnerHTML={{ __html: entry.render }}
dangerouslySetInnerHTML={{ __html: myplugin_custom_sanitization(entry.render) }}
```

Passing in a second parameter into `DOMPurify.sanitize()` will need manual inspection to ensure that the white/blacklists aren't being modified:

```
dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(entry.render, { /*custom settings here*/ }) }}
```

## What could should *not* be reported as a violation?

Only passing in one parameter, we shouldn't need to flag it:

```
dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(entry.render) }}
```

Also thanks to @davidsword for helping me come up with these examples.

**Comment by GaryJones** [comment]
As the JavaScript sniffs have been deprecated for VIPCS 3.1.0, I'm closing this out.
